docs/asciidoc/faq.adoc

```diff
@@ -206,4 +206,102 @@ processor.publishOn(scheduler1)
          .subscribe();
 ----
 
+== What is a good pattern for contextual logging? (MDC)
+As we saw in section <<context>>, `ThreadLocal` doesn't make much sense in Reactive Streams, where several streams could execute steps on a common thread.
+Instead of using `ThreadLocal`, Reactor offers the `Context`, a map-like object that is tied to a `Subscription` and not a `Thread`.
+
+One recurring use of `ThreadLocal` in java is for logging frameworks to store variables that will be used in their logging pattern, generally by way of a `Map` called the MDC ("Mapped Diagnostic Context").
+
+Unfortunately, this pattern is hard to opt out of and hard to apply transparently from a reactive library perspective.
+This entry of the FAQ offers a possible intermediate solution when one wants to perform log statements in relation to events in the stream (`onNext`, `onError`, `onComplete`).
+
+The most straightforward way to perform such logging and get contextual information from the `Context` to the MDC is to wrap logging statements in a `doOnEach` operator with a little bit of boilerplate code.
+This boilerplate depends on both the logging framework/abstraction of your choice and the information you want to put in the MDC, so it has to be in your codebase.
+
+The following is an example of such a helper function around a single MDC variable and focused on logging `onNext` events, using Java 9 enhanced `Optional` API:
+[source,java]
```

Comment:
Above the "[source,java]" line, add another line with four equal signs ("===="). We want to contain the code callouts within a bounded box.